src/zenml/integrations/kubeflow/flavors/kubeflow_orchestrator_flavor.py

```diff
@@ -116,6 +121,37 @@ def _migrate_pod_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             values["node_affinity"] = {}
             values["node_selectors"] = {}
 
+        client_creds_error = AssertionError(
+            "`client_username` and `client_password` both need to be set."
+        )
+        if (
```

Comment:
There are a lot of `values.get("client_username")` and `values.get("client_password")` down below. I would suggest calling it once up here and then using the variables:
```python
username = values.get("client_username")
password = values.get("client_password")

if username and password is None:
  ...
```